### ISSUE TITLE
It would be nice if FastxFile could be initialized from an already open file handle

### ISSUE THREAD

**Issue by terrycojones** [pysam-developers/pysam#525]
Currently `FastxFile` needs a string filename argument. That makes it awkward to replace existing code (e.g., from BioPython) that also works with file handles.

I took a quick look at the source, and it seems the critical lines are in `pysam/libcfaidx.pyx`. There is already special (undocumented?) treatment of filename `-` following the old UNIX convention. Would making this change be as simple a test to see if `filename` is not a string type (and doing what if so?) and being a bit more careful with calling `self.close`.

I'm happy to try implementing this & adding tests, supposing I'm on the right track.

**Comment by AndreasHeger** [comment]
Hi @terrycojones , thanks, good suggestion. This would make FastxFile more consistent with VariantFile and AlignmentFile. If you would like to have a go at this, please do so, otherwise I am happy to do so.

**Comment by terrycojones** [comment]
Hi Andreas

Thanks for the reply. Realistically, I'm not likely to do anything about this in the short term. I was looking for a faster way to read FASTQ than using BioPython and noticed the pysam class. So I gave it a try. It got read time on the file I was working on down to 16 seconds from 24 seconds. But then I tried the BioPython code under pypy and it took 7 seconds!  So I went with that :-)
